From f8a0264cf32b997351ae2e9bf5018dcbd1a377f3 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 6 Oct 2023 17:16:33 -0500
Subject: [PATCH] Call out to sha256sum instead of using hashlib (which
 segfaults)

This fixes an issue where building packages will segfault if the source file is greater than ~3GB or so.

---
 ypkg2/sources.py | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/ypkg2/sources.py b/ypkg2/sources.py
index daaca2e..40a8bc4 100644
--- a/ypkg2/sources.py
+++ b/ypkg2/sources.py
@@ -14,7 +14,6 @@
 from . import console_ui
 
 import os
-import hashlib
 import subprocess
 import fnmatch
 import shutil
@@ -245,12 +244,7 @@ class TarSource(YpkgSource):
     def verify(self, context):
         bpath = self._get_full_path(context)
 
-        hash = None
-
-        with open(bpath, "r") as inp:
-            h = hashlib.sha256()
-            h.update(inp.read())
-            hash = h.hexdigest()
+        hash = subprocess.check_output(["sha256sum", bpath]).split()[0].strip()
         if hash != self.hash:
             console_ui.emit_error("Source", "Incorrect hash for {}".
                                   format(self.filename))
-- 
2.42.0

